pkg/node/types/node.go

```diff
@@ -231,6 +260,9 @@ type Node struct {
 	// node allocates IPs for its local endpoints from
 	IPv6SecondaryAllocCIDRs []*cidr.CIDR
 
+	// IPAMAllocPools ia a list of IPAM pools allocated from the local CiliumNode.
+	IPAMAllocPools []ipamTypes.IPAMPoolAllocation
```

Comment:
Let's not add a field here. `node.Node` is de-facto stable API; it get's serialized to JSON into etcd, so removing fields is difficult. It's also what other nodes consume of this node, and I don't think the `IPAMAllocPools` of this node need to be sent to all other nodes.

Instead, how about adding this field to the `LocalNode` type in `pkg/node/local_node_store.go`.

Reply:
Same feedback as https://github.com/cilium/cilium/pull/27100#discussion_r1316676569.